arangod/Cluster/ClusterComm.cpp

```diff
@@ -332,13 +322,39 @@ void ClusterComm::cleanup() {
 ////////////////////////////////////////////////////////////////////////////////
 
 void ClusterComm::startBackgroundThread() {
-  _backgroundThread = new ClusterCommThread();
+  _roundRobin = 0;
+
+  for(unsigned loop=0; loop<(TRI_numberProcessors()/8+1); ++loop) {
+    ClusterCommThread * thread = new ClusterCommThread();
```

Comment:
We should try to use smart pointers (e.g. `std::unique_ptr` here) rather than raw `new`/`delete` operations.
In case the `push_back` in line 331 fails, the memory of `thread` will leak. This does not matter here, as the server won't start then anyway. But it's good practice anyway to use smart pointers to told "temporary" objects in case something later can go wrong.